flink-end-to-end-tests/test-scripts/test_streaming_kinesis.sh

```diff
@@ -28,9 +28,26 @@ export AWS_SECRET_KEY=flinkKinesisTestFakeAccessKey
 
 KINESALITE_PORT=4567
 
-#docker run -d --rm --name flink-test-kinesis -p ${KINESALITE_PORT}:${KINESALITE_PORT} instructure/kinesalite
-# override entrypoint to enable SSL
-docker run -d --rm --entrypoint "/tini" --name flink-test-kinesis -p ${KINESALITE_PORT}:${KINESALITE_PORT} instructure/kinesalite -- /usr/src/app/node_modules/kinesalite/cli.js --path /var/lib/kinesalite --ssl
+function START_KINESALITE {
+    #docker run -d --rm --name flink-test-kinesis -p ${KINESALITE_PORT}:${KINESALITE_PORT} instructure/kinesalite
+    # override entrypoint to enable SSL
+    docker run -d --rm --entrypoint "/tini" \
+        --name flink-test-kinesis \
+        -p ${KINESALITE_PORT}:${KINESALITE_PORT} \
+        instructure/kinesalite -- \
+        /usr/src/app/node_modules/kinesalite/cli.js --path /var/lib/kinesalite --ssl
+}
+
+function KINESALITE_STARTED {
+    if [[ "$(docker ps --format '{{.Names}}' | grep flink-test-kinesis)" = "flink-test-kinesis" ]]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+START_KINESALITE_MAX_RETRY_NUMBER=10
+retry_until_passed_and_wait_condition START_KINESALITE KINESALITE_STARTED "start kinesalite" ${START_KINESALITE_MAX_RETRY_NUMBER}
```

Comment:
Were you able to reproduce the problem, and does _"retrying"_ mitigate it?

Reply:
I have not looped it on travis, only tested it by turning off and on connection and seeing that it downloads docker image eventually. I can try looping it on Travis but not sure we can do much more against the download hiccups.